mail_restrict_follower_selection/models/mail_followers.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (C) 2018 Creu Blanca
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+from odoo.tools.safe_eval import safe_eval
+
+
+class MailFollowers(models.Model):
+    _inherit = 'mail.followers'
+
+    @api.model
+    def _mail_restrict_follower_selection_get_domain(self):
+        parameter_name = 'mail_restrict_follower_selection.domain'
+        return self.env['ir.config_parameter'].get_param(
+            "{0}.{1}".format(parameter_name,
+                             self.env.context.get('default_res_model')),
+            self.env['ir.config_parameter'].get_param(
+                parameter_name, default='[]')
+        )
+
+    @api.model
+    def _add_follower_command(self, res_model, res_ids, partner_data,
+                              channel_data, force=True):
+        partners = self.env['res.partner'].search(
+            [('id', 'in', list(partner_data))] +
+            safe_eval(self._mail_restrict_follower_selection_get_domain())
+        )
+        return super()._add_follower_command(
+            res_model, res_ids,
+            {p.id: None for p in partners}, channel_data, force)
```

Comment:
you lose the subtypes here. You need to filter the dict to contain only the keys coming back from previous search. And pass `force` as keyword argument to super